src/cmd/services/m3dbnode/config/bootstrap.go

```diff
@@ -43,16 +44,87 @@ import (
 var (
 	// defaultNumProcessorsPerCPU is the default number of processors per CPU.
 	defaultNumProcessorsPerCPU = 0.125
+
+	// default order in which bootstrappers are run
+	// (run in ascending order of precedence).
+	defaultOrderedBootstrappers = []string{
+		// Filesystem bootstrapping must be first.
+		bfs.FileSystemBootstrapperName,
+		// Peers and commitlog must come before the uninitialized topology bootrapping.
+		commitlog.CommitLogBootstrapperName,
+		peers.PeersBootstrapperName,
+		uninitialized.UninitializedTopologyBootstrapperName,
+	}
+
+	// bootstrapper order where peers is prefered over commitlog.
+	preferPeersOrderedBootstrappers = []string{
+		// Filesystem bootstrapping must be first.
+		bfs.FileSystemBootstrapperName,
+		// Prefer peers over commitlog.
+		peers.PeersBootstrapperName,
+		commitlog.CommitLogBootstrapperName,
+		uninitialized.UninitializedTopologyBootstrapperName,
+	}
+
+	validBootstrapModes = []BootstrapMode{
+		DefaultBootstrapMode,
+		PreferPeersBootstrapMode,
+	}
+
+	errReadBootstrapModeInvalid = errors.New("bootstrap mode invalid")
+)
+
+// BootstrapMode defines the mode in which bootstrappers are run.
+type BootstrapMode uint
+
+const (
+	// DefaultBootstrapMode executes bootstrappers in default order.
+	DefaultBootstrapMode BootstrapMode = iota
+	// PreferPeersBootstrapMode executes peers before commitlog bootstrapper.
+	PreferPeersBootstrapMode
 )
 
+// UnmarshalYAML unmarshals an BootstrapMode into a valid type from string.
+func (m *BootstrapMode) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var str string
+	if err := unmarshal(&str); err != nil {
+		return err
+	}
+
+	// If unspecified, use default mode.
+	if str == "" {
+		*m = DefaultBootstrapMode
+		return nil
+	}
+
+	for _, valid := range validBootstrapModes {
+		if str == valid.String() {
+			*m = valid
+			return nil
+		}
+	}
+	return fmt.Errorf("invalid BootstrapMode '%s' valid types are: %s",
+		str, validBootstrapModes)
+}
+
+// String returns the bootstrap mode as a string
+func (m BootstrapMode) String() string {
+	switch m {
+	case DefaultBootstrapMode:
+		return "default"
+	case PreferPeersBootstrapMode:
+		return "prefer_peers"
+	}
+	return "default"
```

Comment:
Would use "unknown" here so it's not confusing if someone ever ends up actually debugging this.